packages/sdk/src/build/build-metadata.js

```diff
@@ -0,0 +1,19 @@
+import {pipe, Ok} from "../interaction/interaction.js"
+
+const ALLOWED_META_FIELDS = new Set(["title", "description", "price", "image"])
+
+export function meta(data = {}) {
+  return pipe([
+    ix => {
+      if (Object.keys(data).length > ALLOWED_META_FIELDS.length) {
+        throw new Error("Wrong number of metadata keys")
+      }
+      Object.keys(data).forEach(
+        key => ALLOWED_META_FIELDS.has(key) || delete data[key]
+      )
+
+      ix.metadata = {...ix.metadata, ...data}
+      return Ok(ix)
+    },
+  ])
```

Comment:
I think we can drop the `pipe` and the `Ok(ix)` (returning just the `ix`).
Im also okay with dropping the `Object.keys(data).length` check. especially if we are scrubbing (and warning) fields.

Reply:
Removed key count check and Ok()

Reply:
Removed pipe and added warning on key scrub